symbolic-debuginfo/src/dwarf.rs

```diff
@@ -529,20 +529,28 @@ impl<'d, 'a> DwarfUnit<'d, 'a> {
     }
 
     /// Parses the call site and range lists of this Debugging Information Entry.
-    fn parse_ranges(
+    ///
+    /// This method consumes the attributes of the DIE. This means that the `entries` iterator must
+    /// be placed just before the attributes of the DIE. On return, the `entries` iterator is placed
+    /// after the attributes, ready to read the next DIE's abbrev.
+    fn parse_ranges<'r>(
         &self,
-        entry: &Die<'d, '_>,
-        range_buf: &mut Vec<Range>,
-    ) -> Result<(Option<u64>, Option<u64>), DwarfError> {
-        let mut tuple = (None, None);
+        entries: &mut gimli::EntriesRaw<gimli::read::EndianSlice<'d, Endian>>,
```

Comment:
the type could probably use a shorter type alias ;-)

Reply:
Good point.

Reply:
Done.